deepchecks/tabular/checks/integrity/outlier_detection.py

```diff
@@ -0,0 +1,149 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Outlier detection functions."""
+from typing import Union, List
+
+import numpy as np
+from PyNomaly import loop
+from sklearn.neighbors import NearestNeighbors
+
+from deepchecks.core import CheckResult, ConditionResult, ConditionCategory
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.utils import gower_distance
+from deepchecks.utils.dataframes import select_from_dataframe
+from deepchecks.utils.typing import Hashable
+
+__all__ = ['OutlierDetection']
+
+
+class OutlierDetection(SingleDatasetCheck):
```

Comment:
The name is a bit ambigious. If we go by the naming of the drift checks, it should be something like "WholeDatasetOutliers"

Reply:
Technically correct, but I must admit I've grown to dislike the "Whole" thing....
Maybe OutlierSampleDetection?

Reply:
Works for me :)